bitwuzla/src/bitwuzla_solver.cpp

```diff
@@ -199,17 +192,21 @@ UnorderedTermMap BzlaSolver::get_array_values(const Term & arr,
 
 void BzlaSolver::get_unsat_assumptions(UnorderedTermSet & out)
 {
-  size_t size;
-  const BitwuzlaTerm ** bcore = bitwuzla_get_unsat_assumptions(bzla, &size);
-  for (size_t i = 0; i < size; ++i)
+  try
+  {
+    std::vector<bitwuzla::Term> bcore = get_bitwuzla()->get_unsat_assumptions();
+    for (auto && elt : bcore)
+    {
+      out.insert(std::make_shared<BzlaTerm>(elt));
+    }
+  }
+  catch (std::exception & e)
```

Comment:
@samanthaarcher0 why did you wrap this in a try-catch?

Reply:
Ah wait I see now, we are wrapping internal solver errors in our own `InternalSolverException`. Maybe we should do this everywhere.